analyzeme/src/profiling_data.rs

```diff
@@ -434,6 +437,76 @@ mod tests {
         }
     }
 
+    #[test]
+    fn can_read_v7_profdata_files() {
+        let (data, file_format_version) = read_data_and_version("tests/profdata/v7.mm_profdata.gz");
+        assert_eq!(file_format_version, file_formats::v7::FILE_FORMAT);
+        let profiling_data = ProfilingData::from_paged_buffer(data, None).expect("Creating the profiling data failed");
+        let grouped_events = group_events(&profiling_data);
+
+        assert_eq!(grouped_events["GenericActivity"].len(), 6428);
+        assert_eq!(grouped_events["IncrementalResultHashing"].len(), 2235);
+        assert_eq!(grouped_events["Query"].len(), 2259);
+
+        assert_eq!(&*grouped_events["GenericActivity"][6428 / 2].label, "incr_comp_encode_dep_graph");
+        assert_eq!(grouped_events["GenericActivity"][6428 / 2].duration(), Some(Duration::from_nanos(0)));
+
+        assert_eq!(&*grouped_events["IncrementalResultHashing"][2235 - 1].label, "item_children");
+        assert_eq!(grouped_events["IncrementalResultHashing"][2235 - 1].duration(), Some(Duration::from_nanos(1300)));
+
+        assert_eq!(&*grouped_events["Query"][0].label, "hir_crate");
+        assert_eq!(grouped_events["Query"][0].duration(), Some(Duration::from_nanos(763800)));
+
+    }
+
+    #[test]
+    fn can_read_v8_profdata_files() {
+        let (data, file_format_version) = read_data_and_version("tests/profdata/v8.mm_profdata.gz");
+        assert_eq!(file_format_version, file_formats::v8::FILE_FORMAT);
+        let profiling_data = ProfilingData::from_paged_buffer(data, None).expect("Creating the profiling data failed");
+        let grouped_events = group_events(&profiling_data);
+
+        assert_eq!(grouped_events["GenericActivity"].len(), 6429);
+        assert_eq!(grouped_events["IncrementalResultHashing"].len(), 2235);
+        assert_eq!(grouped_events["Query"].len(), 2258);
+
+        assert_eq!(&*grouped_events["GenericActivity"][6429 / 2].label, "metadata_decode_entry_implementations_of_trait");
+        assert_eq!(grouped_events["GenericActivity"][6429 / 2].duration(), Some(Duration::from_nanos(1200)));
+
+        assert_eq!(&*grouped_events["IncrementalResultHashing"][2235 - 1].label, "item_children");
+        assert_eq!(grouped_events["IncrementalResultHashing"][2235 - 1].duration(), Some(Duration::from_nanos(213003100)));
```

Comment:
`Duration::from_nanos(213003100)` -- is that number correct? 😮

Reply:
Looks like it. From mmview:

```
{
    kind: IncrementalResultHashing,
    label: item_children,
    additional_data: [],
    payload: 26415 μs - 239418 μs,
    thread_id: 2,
}
```

Is there some large incremental hashing job that happens near the end of the compilation?